tools/release-checks/check-framework-peer-dependency.mts

```diff
@@ -23,11 +23,19 @@ const bzlConfigPath = join(currentDir, '../../packages.bzl');
  */
 export async function assertValidFrameworkPeerDependency(newVersion: semver.SemVer) {
   const currentVersionRange = _extractAngularVersionPlaceholderOrThrow();
-  const isMajorWithPrerelease =
-    newVersion.minor === 0 && newVersion.patch === 0 && !!newVersion.prerelease[0];
-  const requiredRange = isMajorWithPrerelease
-    ? `^${newVersion.major}.0.0-0 || ^${newVersion.major + 1}.0.0`
-    : `^${newVersion.major}.0.0 || ^${newVersion.major + 1}.0.0`;
+  const isMinor = newVersion.minor !== 0;
+  const isPrerelease = !!newVersion.prerelease[0];
+  let requiredRange: string;
+
+  if (isPrerelease) {
+    requiredRange = isMinor
+      ? `^${newVersion.major}.0.0 || ^${newVersion.major}.${newVersion.minor}.0-0 || ^${
+          newVersion.major + 1
+        }.0.0`
+      : `^${newVersion.major}.0.0-0 || ^${newVersion.major + 1}.0.0`;
+  } else {
+    requiredRange = `^${newVersion.major}.0.0 || ^${newVersion.major + 1}.0.0`;
+  }
```

Comment:
Logic looks reasonable to me (just note the extra churn comment). If you have time, would be great to update the comment above to reflect this?

Reply:
I wasn't sure which part of the comment you were referring to with "churn", but I added an example for prerelease minor versions.